### PR TITLE
Add enablehud permission. Fixes: #1149

### PR DESCRIPTION
**Pull request by thegrb93**

### CODE REVIEW

**Comment by thegrb93** [comment]
Could probably do this without the new provider.

**Comment by thegrb93** [comment]
Just needs a test

File: lua/starfall/libs_sh/builtins.lua [hunk 1]
```diff
@@ -8,6 +8,8 @@ local userdataLimit, printBurst
 if SERVER then
 	userdataLimit = CreateConVar("sf_userdata_max", "1048576", { FCVAR_ARCHIVE }, "The maximum size of userdata (in bytes) that can be stored on a Starfall chip (saved in duplications).")
 	printBurst = SF.BurstObject("print", "print", 3000, 10000, "The print burst regen rate in Bytes/sec.", "The print burst limit in Bytes")
+else
+	SF.Permissions.registerPrivilege("enablehud", "Allow enabling hud", "Allows the starfall to enable hud rendering", { client = {} })
```
Comment (x4fx77x4f) [inline lua/starfall/libs_sh/builtins.lua:12]: `{ client = { default = 1 } }`
The default for the permission should be owner-only, not anyone.

**Comment by thegrb93** [comment]
Got it, thx

File: lua/starfall/libs_sh/builtins.lua [hunk 2]
```diff
@@ -1105,7 +1108,7 @@ function builtins_library.enableHud(ply, active)
 	ply = SERVER and getply(ply) or LocalPlayer()
 	checkluatype(active, TYPE_BOOL)
 
-	if ply==instance.player or (instance.player==SF.Superuser) or (not active and SF.IsHUDActive(instance.entity, ply)) then
+	if haspermission(instance, nil, "enablehud") or (not active and SF.IsHUDActive(instance.entity, ply)) then
```
Comment (thegrb93) [inline lua/starfall/libs_sh/builtins.lua:1111]: The permission is client only so the server will need the old code.

**Pull request merged**